src/ripple/core/tests/Config.test.cpp

```diff
@@ -0,0 +1,356 @@
+
+#include <BeastConfig.h>
+#include <ripple/core/impl/LoadFeeTrackImp.h>
+#include <ripple/core/Config.h>
+#include <ripple/basics/TestSuite.h>
+#include <boost/filesystem.hpp>
+#include <boost/format.hpp>
+#include <fstream>
+#include <iostream>
+
+namespace ripple {
+namespace detail {
+std::string configContents (std::string const& dbPath)
+{
+    static boost::format configContentsTemplate (R"rippleConfig(
+[server]
+port_rpc
+port_peer
+port_wss_admin
+
+[port_rpc]
+port = 5005
+ip = 127.0.0.1
+admin = allow
+protocol = https
+
+[port_peer]
+port = 51235
+ip = 0.0.0.0
+protocol = peer
+
+[port_wss_admin]
+port = 6006
+ip = 127.0.0.1
+admin = allow
+protocol = wss
+
+#[port_ws_public]
+#port = 5005
+#ip = 127.0.0.1
+#protocol = wss
+
+#-------------------------------------------------------------------------------
+
+[node_size]
+medium
+
+# This is primary persistent datastore for rippled.  This includes transaction
+# metadata, account states, and ledger headers.  Helpful information can be
+# found here: https://ripple.com/wiki/NodeBackEnd
+# delete old ledgers while maintaining at least 2000. Do not require an
+# external administrative command to initiate deletion.
+[node_db]
+type=memory
+path=/Users/dummy/ripple/config/db/rocksdb
+open_files=2000
+filter_bits=12
+cache_mb=256
+file_size_mb=8
+file_size_mult=2
+
+%1%
+%2%
+
+# This needs to be an absolute directory reference, not a relative one.
+# Modify this value as required.
+[debug_logfile]
+/Users/dummy/ripple/config/log/debug.log
+
+[sntp_servers]
+time.windows.com
+time.apple.com
+time.nist.gov
+pool.ntp.org
+
+# Where to find some other servers speaking the Ripple protocol.
+#
+[ips]
+r.ripple.com 51235
+
+# The latest validators can be obtained from
+# https://ripple.com/ripple.txt
+#
+[validators]
+n949f75evCHwgyP4fPVgaHqNHxUVN15PsJEZ3B3HnXPcPjcZAoy7    RL1
+n9MD5h24qrQqiyBC8aeqqCWvpiBiYQ3jxSr91uiDvmrkyHRdYLUj    RL2
+n9L81uNCaPgtUJfaHh89gmdvXKAmSt5Gdsw2g1iPWaPkAHW5Nm4C    RL3
+n9KiYM9CgngLvtRCQHZwgC2gjpdaZcCcbt3VboxiNFcKuwFVujzS    RL4
+n9LdgEtkmGB9E2h3K4Vp7iGUaKuq23Zr32ehxiU8FWY7xoxbWTSA    RL5
+
+# Ditto.
+[validation_quorum]
+3
+
+# Turn down default logging to save disk space in the long run.
+# Valid values here are trace, debug, info, warning, error, and fatal
+[rpc_startup]
+{ "command": "log_level", "severity": "warning" }
+
+# Defaults to 1 ("yes") so that certificates will be validated. To allow the use
+# of self-signed certificates for development or internal use, set to 0 ("no").
+[ssl_verify]
+0
+
+[sqdb]
+backend=sqlite
+)rippleConfig");
+
+    if (!dbPath.empty ())
+        return boost::str (configContentsTemplate % "[database_path]" % dbPath);
+    else
+        return boost::str (configContentsTemplate % "" % "");
+}
+
+/**
+   Write a config file and remove when done.
+ */
+class ConfigGuard
+{
+private:
+    using path = boost::filesystem::path;
+    path subDir_;
+    path configFile_;
+    path dataDir_;
+
+    bool rmSubDir_{false};
```

Comment:
If `ConfigGuard` is a private class specific to one unit test, it doesn't need a Javadoc comment, a regular comment suffices.